components/ruby/lib/chef-licensing/license_key_fetcher/file.rb

```diff
@@ -208,9 +214,23 @@ def read_license_key_file
         # only checking for major version for file format for breaking changes
         @contents ||= YAML.load(::File.read(path))
         if major_version(@contents[:file_format_version]) == major_version(LICENSE_FILE_FORMAT_VERSION)
+          current_version_class_name = Object.const_get("ChefLicensing::LicenseKeyFetcher::V#{major_version(LICENSE_FILE_FORMAT_VERSION)}LicenseFile")
```

Comment:
this can be a method in itself and used at both places

Reply:
agreed!